src/ussd_nurse_rapidpro.js

```diff
@@ -370,7 +371,161 @@ go.app = function() {
              });
         });
 
+        //USER CHANGE STATES
+        self.states.add('state_change_faccode', function(name) {
+            var question = $("Please enter the 6-digit facility code for your new facility, e.g. 456789:");
+            var error = $("Sorry, that code is not recognized. Please enter the 6-digit facility code again, e. 535970:");
+            return new FreeText(name, {
+                question: question,
+                check: function(content) {
+                    return self
+                        .validate_nc_clinic_code(self.im, content)
+                        .then(function(facname) {
+                            if (!facname) {
+                                return error;
+                            } else {
+                                /*
+                                    add function to update info against operator id
+                                    then lead to state where user details are successfully changed
+                                */
+                                self.im.user.set_answer("facname", facname);
+                                return null;  // vumi expects null or undefined if check passes
+                            }
+                        });
+                },
+                next: 'state_end_detail_changed',
+            });
+               
+        });
+
+        self.states.add('state_change_sanc', function(name) {
+            var question = $("Please enter your 8-digit SANC registration number, e.g. 34567899:");
+            var error = $("Sorry, the format of the SANC registration number is not correct. Please enter it again, e.g. 34567899:");
+            return new FreeText(name, {
+                question: question,
+                check: function(content) {
+                    if (!utils.check_valid_number(content)
+                        || content.length !== 8) {
+                        return error;
+                    } else {
+                        /*
+                            add function to update info against operator id
+                            then lead to state where user details are successfully changed
+                        */
+                        self.im.user.set_answer("sanc_reg_no", content);
```

Comment:
This will already be saved under `state_change_sanc`, so we don't need to save it here again.

Reply:
I think the comments are fine. We need some reminder here that we need to send this update to RapidPro

Reply:
okay thanks